components/identities/updateProfilePic.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+import styles from "../../styles/components/profilePic.module.css";
+import ArrowLeftIcon from "../UI/iconsComponents/icons/arrowLeftIcon";
+import theme from "../../styles/theme";
+import { hexToDecimal } from "../../utils/feltService";
+import NftCard from "../UI/nftCard";
+import ModalProfilePic from "../UI/modalProfilePic";
+
+type UpdateProfilePicProps = {
+  identity?: Identity;
+  tokenId: string;
+  back: () => void;
+  openTxModal: () => void;
+  setPfpTxHash: (hash: string) => void;
+};
+
+const UpdateProfilePic: FunctionComponent<UpdateProfilePicProps> = ({
+  tokenId,
+  identity,
+  back,
+  openTxModal,
+  setPfpTxHash,
+}) => {
+  const [userNft, setUserNft] = useState<StarkscanNftProps[]>([]);
+  const [openModal, setOpenModal] = useState<boolean>(false);
+  const [selectedPic, setSelectedPic] = useState<StarkscanNftProps | null>(
+    null
+  );
+  const whitelistedContracts = [
+    // Starknet Quest contracts
+    hexToDecimal(
+      "0x0154520b48b692bb8b926434bbd24d797e806704af28b6cdcea30ea7db6a996b"
+    ),
+    // Identicons
+    hexToDecimal(
+      "0x0783a9097b26eae0586373b2ce0ed3529ddc44069d1e0fbc4f66d42b69d6850d"
+    ),
+    // Braavos shield
+    hexToDecimal(
+      "0x04353bb6424de0b468ec4c984e01637fb2fafd3bdf81f4af367077fcbb9382c1"
+    ),
+    // Briq
+    hexToDecimal(
+      "0x01435498bf393da86b4733b9264a86b58a42b31f8d8b8ba309593e5c17847672"
+    ),
+    // Eykar
+    hexToDecimal(
+      "0x041e1382e604688da7f22e7fbb6113ba3649b84a87b58f4dc1cf5bfa96dfc2cf"
+    ),
+  ];
```

Comment:
This is testnet ? Mainnet ? Cause hardcoding will be difficult to manage.

Also I think we need to do something cleaner on another folder as we did for `registerCallData`

Reply:
We also have to discuss on who to whitelist, maybe an ENV variable as an array can be better

Reply:
Yes, I forgot to update it, so it's not possible to have an array of string as env variable. I tried different things like having a string of addresses that I could split, or a string that I could parse with JSON.parse but it didn't work. For now I've added multiple env variables and then I build a string array from those env variables.